### ISSUE TITLE
Z-coordinate difference between Height service and embedded values in swissBUILDINGS3D 2.0

### ISSUE THREAD

**Issue by RobertoMinelli** [geoadmin/mf-chsdi3#3156]
Dear all,

I am trying to create a single 3D model (`.stl`) that contains both a 3D surface representing a terrain and the buildings that lie on that terrain.

To generate the **terrain surface**, I create a XY-grid of coordinates and I obtain the elevation using the [Height service](https://api3.geo.admin.ch/services/sdiservices.html#height) of the [GeoAdmin API REST services](https://api3.geo.admin.ch/services).

I obtain the **buildings** contained in the terrain by leveraging the [swissBUILDINGS3D 2.0 dataset](https://shop.swisstopo.admin.ch/en/products/landscape/build3D2) that I've bought.

Now, the **problem**: The `COLLADA` (`.dae`) models contained in the swissBUILDINGS3D 2.0 dataset have embedded XYZ coordinates (i.e., each building knows where it should be placed on the terrain). X and Y coordinates match with my terrain. Unfortunately, the Z-coordinate do not march (i.e., the terrain is always above the buildings, as per screenshot attached). Do you have any intuition? Am I doing something wrong?

![screenshot_2019-04-05_at_11 20 17](https://user-images.githubusercontent.com/6087804/56197312-dafd9e00-6038-11e9-9903-a759d47373cc.png)

cc: @danduk82 @davidoesch @oterral @procrastinatio 😇

**Comment by procrastinatio** [comment]
I am not too familiar with the height issue, but I guess it is because the `height` service and `SwissBuilding`  are using the _usual height_  ([LN02](https://www.swisstopo.admin.ch/en/knowledge-facts/surveying-geodesy/reference-frames/local/ln02.html)) and Collada is assuming it is the _WGS84 ellipsoidal height_, hence the about 50 meters discrepancy.
All local (=swiss) _systems_ (LN02, RAN95, etc.) are usually within a few meters margins.

**Comment by RobertoMinelli** [comment]
Hi Marc,

Thanks for your answer. Are you sure that the swissBuildings3D dataset (i.e., Collada) is using the WGS84 ellipsoidal height? Why are the X and Y coordinates correct (i.e., in Swiss coordinate system).

**Comment by RobertoMinelli** [comment]
Thanks, @procrastinatio. You are right, Collada uses WGS84 🤦‍♂️

**Comment by danduk82** [comment]
@RobertoMinelli is this still an issue?

**Comment by RobertoMinelli** [comment]
Yes, @danduk82. When I assemble a terrain obtained by triangulating 3D points obtained using the `Height Service` the result is always what is in the first picture. Moreover, If I move uniformly the buildings to touch the terrain, when the building with the lowest Z-coordinate touches the ground, buildings on mountains fly. Do you have any workaround for that?

**Comment by danduk82** [comment]
@RobertoMinelli, you should convert your colladata altitudes to [LN02](https://www.swisstopo.admin.ch/en/knowledge-facts/surveying-geodesy/reference-frames/local/ln02.html).

As @procrastinatio told here above, in our data, the Z coordinate is not reprojected on the WGS84 ellipsoid, but for "consistency with the original dataset reasons" we keep the Z coordinates that we get in the original LV95 datasets.

**Comment by RobertoMinelli** [comment]
Grazie @danduk82. Any hint on how to read and convert Z-coordinates from `COLLADA` models? As of now I was only converting `COLLADA` to `.stl` in order to merge it with the terrain (obtained bu querying your `Height Service`.

I am just wondering then... why are the altitudes in the `swissBUILDINGS3D` dataset in `WGS84`? Because of the format, i.e., `COLLADA`?

Thank you very much,
Rob

**Comment by davidoesch** [comment]
Yes. Inline/collars is wgs84

Roberto Minelli <notifications@github.com> schrieb am Mi. 15. Mai 2019 um
17:25:

**Comment by RobertoMinelli** [comment]
Any news on this @danduk82? It would be amazing if there was a way to have a match between the two height models... Moreover, as you can see in the screenshots below, if I manually align one of the buildings to the ground, other buildings are _"flying"_. As of now, my terrains have a resolution of 25 meters (i.e., I query the [Height Service](https://api3.geo.admin.ch/services/sdiservices.html#height) every 25 meters in a swipe fashion then I triangulate points to get the surface).

![original](https://user-images.githubusercontent.com/6087804/63603628-50224680-c5ca-11e9-8417-4648835faa5b.png)
Situation right after the import

![shift-z](https://user-images.githubusercontent.com/6087804/63603632-54e6fa80-c5ca-11e9-8a33-83f1fe899716.png)
Situation after shifting up all the buildings and aligning the one with lowest z-coordinate to the ground

![flying](https://user-images.githubusercontent.com/6087804/63603641-56b0be00-c5ca-11e9-8757-f8dae8805def.png)
As you can see, other buildings are flying 😢

I also attach the two .stl models of terrain and buildings.

[STL_models.zip](https://github.com/geoadmin/mf-chsdi3/files/3535130/STL_models.zip)

Thanks,
Roberto
